### PR TITLE
Testnet Mainnet preparation

### PR DESCRIPTION
**Pull request by StarbuckBG**
Working on changing the seeds and rising up new ones with : https://github.com/StarbuckBG/bitcoin-gold-seeder

### CODE REVIEW

**Comment by jayarjo** [comment]
So what exactly is required? I see you did some changes already. Do you need someone to launch some seeders now?

**Comment by jagottsicher** [comment]
@StarbuckBG tried your dnsseeds, but couldn't make any connection to it.

**Comment by StarbuckBG** [comment]
I’m currently working in the DDoS attack on the main site. You need the modified code from my PR.

**Comment by jagottsicher** [comment]
good luck against the ddos.
I used your chainparams.cpp mentioned in the pull request, but without success.

**Comment by StarbuckBG** [comment]
We are rising seeding nodes and DNSseeds. I will text then I’m ready

**Comment by tankitty** [comment]
good

File: src/chainparams.cpp [hunk 1]
```diff
@@ -150,12 +150,12 @@ class CMainParams : public CChainParams {
 
         // Note that of those with the service bits flag, most only support a subset of possible options
         // TODO: Shall change this to our
-        vSeeds.emplace_back("seed.bitcoin.sipa.be", true); // Pieter Wuille, only supports x1, x5, x9, and xd
-        vSeeds.emplace_back("dnsseed.bluematt.me", true); // Matt Corallo, only supports x9
-        vSeeds.emplace_back("dnsseed.bitcoin.dashjr.org", false); // Luke Dashjr
-        vSeeds.emplace_back("seed.bitcoinstats.com", true); // Christian Decker, supports x1 - xf
-        vSeeds.emplace_back("seed.bitcoin.jonasschnelli.ch", true); // Jonas Schnelli, only supports x1, x5, x9, and xd
-        vSeeds.emplace_back("seed.btc.petertodd.org", true); // Peter Todd, only supports x1, x5, x9, and xd
+//        vSeeds.emplace_back("seed.bitcoin.sipa.be", true); // Pieter Wuille, only supports x1, x5, x9, and xd
```
Comment (h4x3rotab) [inline src/chainparams.cpp]: Don't leave commented out code here. Just delete them.

**Comment by jagottsicher** [comment]
How about a new chainsparams.cpp, respectively DNSseed-Server?

**Comment by StarbuckBG** [comment]
I add them manually on the tests. As soon as the network is working I will add them hardcoded.

**Comment by tankitty** [comment]
@StarbuckBG RPC port is same with bitcoind?

**Comment by StarbuckBG** [comment]
Doesn’t matter. You can put to to whatever port.

File: doc/build-unix.md [hunk 2]
```diff
@@ -73,7 +73,7 @@ Options when installing required Boost library files:
 individual boost development packages, so the following can be used to only
 install necessary parts of boost:
 
-        sudo apt-get install libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-program-options-dev libboost-test-dev libboost-thread-dev
+        sudo apt-get install libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-program-options-dev libboost-test-dev libboost-thread-dev libboost1.58-all-dev
```
Comment (vattay) [inline doc/build-unix.md]: Doesn't libboost1.58-all-dev include all the previous boost libs?

File: src/net.h [hunk 3]
```diff
@@ -692,7 +697,11 @@ class CNode
     std::atomic<int64_t> nMinPingUsecTime;
     // Whether a ping is requested.
     std::atomic<bool> fPingQueued;
+    // Whether the node uses the bitcoin cash magic to communicate.
```
Comment (vattay) [inline src/net.h]: bitcoin cash?

**Comment by StarbuckBG** [comment]
Lol will fix it :D I have being checking their logic at some point for this.

File: test/functional/p2p-acceptblock.py [hunk 4]
```diff
@@ -57,7 +57,7 @@
 class AcceptBlockTest(BitcoinTestFramework):
     def add_options(self, parser):
         parser.add_option("--testbinary", dest="testbinary",
-                          default=os.getenv("BITCOIND", "bitcoind"),
+                          default=os.getenv("BITCOIND", "bgoldd"),
                           help="bitcoind binary to test")
```
Comment (vattay) [inline test/functional/p2p-acceptblock.py]: "bgoldd binary to test"

File: src/pow.cpp [hunk 5]
```diff
@@ -75,10 +76,12 @@ unsigned int CalculateNextWorkRequired(arith_uint256 bnAvg, int64_t nLastBlockTi
 unsigned int BitcoinGetNextWorkRequired(const CBlockIndex* pindexLast, const CBlockHeader *pblock, const Consensus::Params& params)
 {
     assert(pindexLast != nullptr);
-    unsigned int nProofOfWorkLimit = UintToArith256(params.PowLimit(false)).GetCompact();
-
-    // Only change once per difficulty adjustment interval
-    if ((pindexLast->nHeight+1) % params.DifficultyAdjustmentInterval() != 0)
+    unsigned int nProofOfWorkLimit = UintToArith256(params.powLimit).GetCompact();
```
Comment (h4x3rotab) [inline src/pow.cpp]: The change of this function should be reverted. I just changed it back to the same as Bitcoin Core.

File: src/validation.cpp [hunk 6]
```diff
@@ -3120,6 +3120,7 @@ bool ProcessNewBlockHeaders(const std::vector<CBlockHeader>& headers, CValidatio
     return true;
 }
 
+
```
Comment (h4x3rotab) [inline src/validation.cpp]: Remove the empty line.

File: src/net_processing.cpp [hunk 7]
```diff
@@ -2723,16 +2726,25 @@ bool ProcessMessages(CNode* pfrom, CConnman& connman, const std::atomic<bool>& i
...
 
     msg.SetVersion(pfrom->GetRecvVersion());
+    
+    // This is a new peer. Before doing anything, we need to detect what magic
+    // the peer is using.
+    if (pfrom->nVersion == 0 &&
+        memcmp(msg.hdr.pchMessageStart, chainparams.MessageStart(),
+               CMessageHeader::MESSAGE_START_SIZE) == 0) {
+            pfrom->fUsesGoldMagic = true;
+        }
```
Comment (h4x3rotab) [inline src/net_processing.cpp]: Wrong indent. Please remove 4 spaces.

File: src/pow.cpp [hunk 8]
```diff
@@ -16,6 +16,7 @@
 
 unsigned int GetNextWorkRequired(const CBlockIndex* pindexLast, const CBlockHeader *pblock, const Consensus::Params& params)
 {
+
```
Comment (h4x3rotab) [inline src/pow.cpp]: Remove the empty line.

File: src/net.h [hunk 9]
```diff
@@ -638,6 +640,9 @@ class CNode
     const uint64_t nKeyedNetGroup;
     std::atomic_bool fPauseRecv;
     std::atomic_bool fPauseSend;
+
+    const NodeId id;
```
Comment (h4x3rotab) [inline src/net.h]: Is it used somewhere?

File: README.md [hunk 10]
```diff
@@ -13,5 +13,9 @@ Bitcoin Gold is a work in progress. If you don’t understand what you are doing
 ## Links
 
 * Website: http://btcgpu.org
+<<<<<<< HEAD
```
Comment (h4x3rotab) [inline README.md]: Revert this file

File: src/net.h (hunk 9 above)
Comment (StarbuckBG) [reply src/net.h]: it is required in derived class as much as I remember

**Pull request merged**